test/prog/modes/README

```diff
@@ -0,0 +1,92 @@
+AUTOTEST
+========
```

Comment:
How about converting this one to reStructuredText, even if it is only for the nice preview on GH?

Reply:
Done. Of course it means the DFTB+ test README also needs a PR.